### PR TITLE
Support bare pod move in the integration test

### PR DESCRIPTION
**Pull request by kevinwangcn**
# Intent
Adding move action execution test for the bare pod

# Background
The current integration test doesn't include this case

# Implementation
Create a bare pod and move it from one node to another node

# Test Done

## Case 1
**1. Build the binary and run the newly added case by executing this command**
```
make integration
./build/integration.test -k8s-kubeconfig=/root/.kube/config -k8s-context=kind-kind -ginkgo.focus="executing action move bare pod"
```
image

### CODE REVIEW

File: test/integration/action_execution.go [hunk 1]
```diff
@@ -314,6 +334,68 @@ func depSingleContainerWithResources(namespace, claimName string, replicas int32
...
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return waitForBarePod(client, barepod.Name, barepod.Namespace)
+}
+
+func depBarePodWithResources(namespace, claimName string, replicas int32, withVolume bool) *corev1.Pod {
```
Comment (irfanurrehman) [inline test/integration/action_execution.go:354]: What does `dep` prefix mean here?

File: test/integration/action_execution.go [hunk 2]
```diff
@@ -314,6 +334,68 @@ func depSingleContainerWithResources(namespace, claimName string, replicas int32
...
+							corev1.ResourceCPU:    resource.MustParse("50m"),
+							corev1.ResourceMemory: resource.MustParse("100Mi"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if withVolume {
```
Comment (irfanurrehman) [inline test/integration/action_execution.go:383]: Why not add a test for volume based bare pod moves also, given you have already added the code to have a volume with the pod.

File: test/integration/action_execution.go [hunk 3]
```diff
@@ -643,7 +744,7 @@ func getDeploymentConfigsPod(client kubeclientset.Interface, dcName, namespace,
 		}
 	}
 
-	return nil, nil
+	return nil, fmt.Errorf("Can't find the right pod")
```
[the remark this answers is not in the document]
Comment (kevinwangcn) [reply test/integration/action_execution.go]: done

File: test/integration/action_execution.go [hunk 4]
```diff
@@ -659,7 +760,7 @@ func validateMovedPod(client kubeclientset.Interface, parentName, parentType, na
 	var pod *corev1.Pod
 	var err error
 
-	if parentType == "deployment" {
+	if parentType == "deployment" || parentType == "barepod" {
 		pod, err = getDeploymentsPod(client, parentName, namespace, targetNodeName)
```
[the remark this answers is not in the document]
Comment (kevinwangcn) [reply test/integration/action_execution.go]: change it to `getPodWithNamePrefix`

File: test/integration/action_execution.go (hunk 1 above)
Comment (kevinwangcn) [reply test/integration/action_execution.go]: I think it means `deploy`,  anyway, I changed the function name to `genBarePodWithResources`

File: test/integration/action_execution.go [hunk 5]
```diff
@@ -314,6 +334,68 @@ func depSingleContainerWithResources(namespace, claimName string, replicas int32
...
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return waitForBarePod(client, barepod.Name, barepod.Namespace)
+}
+
+func genBarePodWithResources(namespace, claimName string, replicas int32, withVolume bool) *corev1.Pod {
```
Comment (pankajbatra1) [inline test/integration/action_execution.go:354]: You could refactor portions of `genBarePodWithResources ` , `depSingleContainerWithResources`, etc.  There is overlap.

Comment (pankajbatra1) [reply test/integration/action_execution.go:354]: Resolving since this is covered by the deploymentconfig pr

**Pull request merged**